facebook-commerce.php

```diff
@@ -597,81 +597,73 @@ public function get_graph_api() {
 	 * @return array
 	 */
 	public function get_variation_product_item_ids( $product, $product_group_id ) {
-
-		$product_item_ids_by_variation_id = array();
-		$missing_product_item_ids         = array();
-
+		$ids_by_variation_id = array();
+		$missing_ids         = array();
 		// get the product item IDs from meta data and build a list of variations that don't have a product item ID stored
 		foreach ( $product->get_children() as $variation_id ) {
-
 			if ( $variation = wc_get_product( $variation_id ) ) {
-
 				if ( $product_item_id = $variation->get_meta( self::FB_PRODUCT_ITEM_ID ) ) {
-
-					$product_item_ids_by_variation_id[ $variation_id ] = $product_item_id;
-
+					$ids_by_variation_id[ $variation_id ] = $product_item_id;
 				} else {
-
-					$retailer_id = WC_Facebookcommerce_Utils::get_fb_retailer_id( $variation );
-
-					$missing_product_item_ids[ $retailer_id ] = $variation;
-
-					$product_item_ids_by_variation_id[ $variation_id ] = null;
+					$retailer_id                          = WC_Facebookcommerce_Utils::get_fb_retailer_id( $variation );
+					$missing_ids[ $retailer_id ]          = $variation;
+					$ids_by_variation_id[ $variation_id ] = null;
 				}
 			}
 		}
-
 		// use the Graph API to try to find and store the product item IDs for variations that don't have a value yet
-		if ( $missing_product_item_ids ) {
-
-			$product_item_ids = $this->find_variation_product_item_ids( $product_group_id );
-
-			foreach ( $missing_product_item_ids as $retailer_id => $variation ) {
-
-				if ( isset( $product_item_ids[ $retailer_id ] ) ) {
-
-					$variation->update_meta_data( self::FB_PRODUCT_ITEM_ID, $product_item_ids[ $retailer_id ] );
+		if ( $missing_ids ) {
+			$ids = $this->find_variation_product_item_ids( $product_group_id );
+			foreach ( $missing_ids as $retailer_id => $variation ) {
+				if ( isset( $ids[ $retailer_id ] ) ) {
+					$variation->update_meta_data( self::FB_PRODUCT_ITEM_ID, $ids[ $retailer_id ] );
 					$variation->save_meta_data();
-
-					$product_item_ids_by_variation_id[ $variation->get_id() ] = $product_item_ids[ $retailer_id ];
+					$ids_by_variation_id[ $variation->get_id() ] = $ids[ $retailer_id ];
 				}
 			}
 		}
-
-		return $product_item_ids_by_variation_id;
+		return $ids_by_variation_id;
 	}
 
 
 	/**
 	 * Uses the Graph API to return a list of Product Item IDs indexed by the variation's retailer ID.
-	 *
-	 * @since 2.0.0
+	 * 	Returns a map of pairs
+	 * 	e.g.
+	 * 	(
+	 * 		`woo-vneck-tee-blue_28` -> `7344216055651160`,
+	 * 		`woo-vneck-tee-red_26`  -> `5102436146508829`
+	 * 	)
 	 *
 	 * @param string $product_group_id product group ID
-	 * @return array
+	 * @return array a map of ( `retailer id` -> `id` ) pairs.
 	 */
-	private function find_variation_product_item_ids( $product_group_id ) {
-
-		$product_item_ids = array();
-
+	private function find_variation_product_item_ids( string $product_group_id ): array {
+		$product_item_ids = [];
```

Comment:
For consistency's sake, we'd probably have to stick to one syntax style for the arrays. Can we change this to array to keep the consistency in this extension perhaps?

PS:  I see we use the short array syntax in some of the other extensions but on this extension, we still use array().  Maybe there's a bigger discussion to be had with regard to standardizing this.

Reply:
Adjusted. Thanks for noticing.